projects/step-frontend/src/lib/modules/timeseries/dashboard/time-series-dashboard.component.ts

```diff
@@ -89,26 +87,23 @@ export class TimeSeriesDashboardComponent implements OnInit, OnDestroy {
 
   private mergeContextualParamsWithActiveFilters(
     contextualParams: Record<string, string>,
-    activeFilters: TsFilterItem[]
+    activeFilters: FilterBarItem[]
   ) {
     const contextualFilters = Object.keys(contextualParams).map((key) => {
       const fieldType = this.getFilterFieldType(key);
-      const isEntityFilter =
-        fieldType === FilterBarItemType.PLAN ||
-        fieldType === FilterBarItemType.TASK ||
-        fieldType === FilterBarItemType.EXECUTION;
+      const isEntityFilter = fieldType === 'PLAN' || fieldType === 'TASK' || fieldType === 'EXECUTION';
       const searchValue = contextualParams[key];
       return {
         isHidden: !this.settings.showContextualFilters,
         removable: true,
         label: this.getFilterFieldLabel(key),
         type: fieldType,
         attributeName: key,
-        freeTextValues: fieldType === FilterBarItemType.FREE_TEXT ? [searchValue] : [],
+        freeTextValues: fieldType === 'FREE_TEXT' ? [searchValue] : [],
```

Comment:
Why was the enum reference removed? Generally I prefer ENUMS to string defined types

Reply:
@neogucky me too! it is generated from the backend now.. if you want we can layer an enum on top of it

Reply:
I'm not sure what you mean, this type is not defined in the generated class but in filter-bar-item.ts: export type FilterBarItemType = 'OPTIONS' | 'FREE_TEXT' | 'EXECUTION' | 'TASK' | 'PLAN' | 'NUMERIC' | 'DATE';

In my opinion you could instead define it like: 
export enum FilterBarItemType {
  OPTIONS = 'OPTIONS',
  FREE_TEXT = 'FREE_TEXT',
  EXECUTION = 'EXECUTION',
  TASK = 'TASK',
  PLAN = 'PLAN',
  NUMERIC = 'NUMERIC',
  DATE = 'DATE',
}

and use it everywhere as an enum and type.

Reply:
@neogucky I forgot I defeined it that way. it is a 1-1 copy of TimeSeriesFilterItem.ts - type property, so normally it should be removed. If you want an enum which maps these strings, I definitely like it

Reply:
Then please switch to enum and use it for the strings, that will be a saver solution for sure!